smtpsrv/server.go

```diff
@@ -2,12 +2,37 @@ package smtpsrv
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	"net"
+	"strings"
 	"time"
 
 	"github.com/emersion/go-smtp"
+	"github.com/target/goalert/util/log"
 )
 
+// SMTPLogger implements the smtp.Logger interface using the main app Logger
+type SMTPLogger struct {
+	logger *log.Logger
+}
+
+// Printf adheres to smtp.Server's Logger interface while filtering out ECONNRESET errors caused by TCP health checks
+func (l *SMTPLogger) Printf(format string, v ...interface{}) {
+	s := fmt.Sprintf(format, v...)
+	if !strings.Contains(s, "read: connection reset by peer") {
+		l.logger.Error(context.Background(), errors.New(s))
+	}
+}
```

Comment:
nit: prefer early return, invert/short circuit stuff like this so the main work happens in the outer body

> The code reads well if the successful flow of control runs down the page, eliminating error cases as they arise.
>
> https://go.dev/doc/effective_go

```suggestion
	if strings.Contains(s, "read: connection reset by peer") {
		return // ignore connection reset for now
	}

	l.logger.Error(context.Background(), errors.New(s))
}
```

---

alternatively the TODO could be put here

Reply:
Good catch, thank you for picking that nit.  Also added the TODO here rather than above the func definition since these functions will still be needed to bring go-smtp logging under the main GoAlert logger.